src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -1119,10 +1120,16 @@ cu_stat(struct sched_cmd *cmd)
 
 	/* indevidual SK CU status */
 	for (i = 0; i < sk->sk_ncus && pkt_idx < max_idx; ++i) {
-		if (sk->sk_cu[i])
-			pkg->data[pkt_idx++] = (exec->scu_status[cu_mask_idx(i)] &
-				(1 << (i % sizeof(exec->scu_status[0])))) ? 1 : 0;
-		else
+		p_tmp = NULL;
+		if (sk->sk_cu[i]) {
+			/* xbutil to check if process is alive */
+			p_tmp = find_get_pid(sk->sk_cu[i]->sc_pid);
```

Comment:
I think this could be a temporary solution for now. Let's revisit this in the next couple of weeks. I think a more robust solution should be introducing softkernel context in zocl. Once the context is gone, sk_cu[i] should be NULL.

However, I do have some concern here, since we are not holding sk_lock, do we have a crash window that sk_cu[i] was just released after line 1124?

Reply:
Updated for sk_lock. Checking process alive is still better as xbutil query can be used anytime even after end of an application. context may have dependency on a running application

Reply:
I mean softkernel process context on device side but not the host side application.